### ISSUE TITLE
If there is no neg column in the training data when training reranker, modify reranker/data.py

### ISSUE THREAD

**Issue by daegonYu** [FlagOpen/FlagEmbedding#1104]
Since there is no neg in the training data, I modified it as shown below. Is it possible that training is not performed properly?
In other words, can there be an error in the learning code due to the absence of neg?

path : FlagEmbedding/FlagEmbedding/reranker/data.py    
```py

    def __getitem__(self, item) -> List[BatchEncoding]:
        query = self.dataset[item]['query']
        pos = random.choice(self.dataset[item]['pos'])
        if 'neg' in self.dataset.column_names:
            if len(self.dataset[item]['neg']) < self.args.train_group_size - 1:
                num = math.ceil((self.args.train_group_size - 1) / len(self.dataset[item]['neg']))
                negs = random.sample(self.dataset[item]['neg'] * num, self.args.train_group_size - 1)
            else:
                negs = random.sample(self.dataset[item]['neg'], self.args.train_group_size - 1)

            batch_data = []
            batch_data.append(self.create_one_example(query, pos))
            for neg in negs:
                batch_data.append(self.create_one_example(query, neg))
                
        else:
            batch_data = []
            batch_data.append(self.create_one_example(query, pos))

        return batch_data

```

**Comment by 545999961** [comment]
`neg` is necessary.
You can min hard negatives following this [example](https://github.com/FlagOpen/FlagEmbedding/tree/master/examples/finetune#hard-negatives)

**Comment by daegonYu** [comment]
Oh, thank you. Is this trained with in-batch negatives loss?

**Comment by 545999961** [comment]
> Oh, thank you. Is this trained with in-batch negatives loss?

There is no need to use in-batch negatives for reranker.
